pkg/controllers/provisioning/scheduling/volumetopology.go

```diff
@@ -153,44 +137,31 @@ func (v *VolumeTopology) getPersistentVolumeRequirements(ctx context.Context, po
 	return requirements, nil
 }
 
-func (v *VolumeTopology) getPersistentVolumeClaim(ctx context.Context, pod *v1.Pod, volume v1.Volume) (*v1.PersistentVolumeClaim, error) {
-	if volume.PersistentVolumeClaim == nil {
-		return nil, nil
-	}
-	pvc := &v1.PersistentVolumeClaim{}
-	if err := v.kubeClient.Get(ctx, types.NamespacedName{Name: volume.PersistentVolumeClaim.ClaimName, Namespace: pod.Namespace}, pvc); err != nil {
-		return nil, fmt.Errorf("getting persistent volume claim %q, %w", volume.PersistentVolumeClaim.ClaimName, err)
-	}
-	return pvc, nil
-}
-
 // ValidatePersistentVolumeClaims returns an error if the pod doesn't appear to be valid with respect to
 // PVCs (e.g. the PVC is not found or references an unknown storage class).
 func (v *VolumeTopology) ValidatePersistentVolumeClaims(ctx context.Context, pod *v1.Pod) error {
 	for _, volume := range pod.Spec.Volumes {
-		var storageClassName *string
-		var volumeName string
-		if volume.PersistentVolumeClaim != nil {
-			// validate the PVC if it exists
-			pvc, err := v.getPersistentVolumeClaim(ctx, pod, volume)
-			if err != nil {
+		pvc, err := volumeutil.GetPersistentVolumeClaim(ctx, v.kubeClient, pod, volume)
+		if err != nil {
+			return err
+		}
+		// may not have a pvc
```

Comment:
Can we callout when it's possible that the volume doesn't have a PVC. IIUC, it's a case of when it's using an in-tree provider that isn't using CSI. In this case, we are just ignoring the volume usage since we have no ability to discover it

Reply:
That's what the comment is calling out on 148, I can be more verbose about it.

Reply:
Sure, I got that, but I still don't know the why around when something doesn't have a PVC

Reply:
Ah I get what you mean, idk what I was thinking when I initially responded. That scenario sounds correct, I can lookup the full list of volume types which don't need a PVC.